add docs on running interactive python apps (#16)
posit-dev/positron-website: run-interactive-apps.qmd

---
title: "Run Interactive Apps"
---

> Positron provides a simplified method for running interactive apps via the **Play** button in the **Editor Actions** area. Instead of running an app from a **Terminal**, you can run supported apps by clicking the **Play** button. You can also start a supported app in debug mode through the **Play** button context menu.

Reviewer: ```suggestion
Positron provides a simplified method for running interactive apps via the **Play** button. Instead of running an app from a **Terminal**, you can run supported apps by clicking the **Play** button in Editor Actions.  Additionally, you can start a supported app in debug mode through the **Play** button context menu.
```

![Positron Run App button](images/run-app-button.png)

## Supported app frameworks

Currently, Positron supports the following Python app frameworks:

- Dash
- FastAPI
- Flask
- Gradio
- Shiny
- Streamlit

## Running an interactive app

1. Open the `.py` file of a supported app framework.

> 2. Click the **Play** button in the **Editor Actions** bar to start the app.

Reviewer: ```suggestion
2. In Editor Actions, click **Play**.
```

    ![Run App Play Button](images/run-app-play-button.png)

> 3. Positron runs the app in a dedicated Terminal tab and opens the app URL in the Viewer Pane.

Reviewer: This shouldn't be a step because it doesn't instruct the user to complete a task.

    ![A Streamlit app in the Positron Viewer](images/run-app-streamlit.png)

## Debugging an interactive app

1. Open the `.py` file of a supported app framework.
2. Set breakpoints in the `.py` file by clicking on the editor margin.
    ![Set breakpoint](images/run-app-debug-breakpoint.png)

> 3. Open the context menu for the **Play** button and select **Debug [*{SUPPORTED_APP_TYPE}*] App in Terminal**.

Reviewer: ```suggestion
3. Click the **Play** button drop-down context menu and select **Debug [*{SUPPORTED_APP_TYPE}*] App in Terminal**.
```

>     - *{SUPPORTED_APP_TYPE}* is one of the supported app frameworks, which is *Streamlit* in this example.

Reviewer: ```suggestion
    - For this example, we select **Debug Steamlit App in Terminal**.
```

    ![Run App Dropdown](images/run-app-debug-streamlit.png)

> 4. Positron runs the app in a dedicated **Terminal** tab, opens the app URL in the **Viewer** pane, and starts the app in debug mode.

Reviewer: This shouldn't be a step.
